Description: fix issue with Worker.pm
 patch to fix Worker connection issue (#611355)
Origin: vendor
Bug-Debian: http://bugs.debian.org/611355
Bug: https://rt.cpan.org/Ticket/Display.html?id=85191
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=85191
Author: Richard Wallman <wallmari@bossolutions.co.uk>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-05-11

--- a/lib/Gearman/Worker.pm
+++ b/lib/Gearman/Worker.pm
@@ -263,6 +263,10 @@
     # ( js => last_update_time, ... )
     my %last_update_time;
 
+    my $on_connect = sub {
+        return Gearman::Util::send_req($_[0], \$presleep_req);
+    };
+
     while (1) {
         # "Jobby" job servers are the set of server which we will contact
         # on this pass through the loop, because we need to clear and use
@@ -278,7 +282,7 @@
         for (my $i = 0; $i < $js_count; $i++) {
             my $js_index = ($i + $js_offset) % $js_count;
             my $js = $jobby_js[$js_index];
-            my $jss = $self->_get_js_sock($js)
+            my $jss = $self->_get_js_sock($js, on_connect => $on_connect)
                 or next;
 
             # TODO: add an optional sleep in here for the test suite
@@ -380,10 +384,6 @@
 
         my @jss;
 
-        my $on_connect = sub {
-            return Gearman::Util::send_req($_[0], \$presleep_req);
-        };
-
         foreach my $js (@{$self->{job_servers}}) {
             my $jss = $self->_get_js_sock($js, on_connect => $on_connect)
                 or next;
